packages/rest/src/router/router-base.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright IBM Corp. 2017, 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {RestRouter, RouteEntry, createResolvedRoute} from './routing-table';
+import {Request} from '../types';
+import {getPathVariables} from './openapi-path';
+
+/**
+ * Base router implementation that only handles path without variables
+ */
+export abstract class BaseRouter implements RestRouter {
+  /**
+   * A map to optimize matching for routes without variables in the path
+   */
+  protected simpleRoutes: {[path: string]: RouteEntry} = {};
+
+  protected getKeyForRoute(route: RouteEntry) {
+    const path = route.path.startsWith('/') ? route.path : `/${route.path}`;
+    const verb = route.verb.toLowerCase() || 'get';
+    return `/${verb}${path}`;
+  }
+
+  add(route: RouteEntry) {
+    const key = this.getKeyForRoute(route);
```

Comment:
Let's avoid preliminary pessimization and obtain the key only if we are going to register this route ourselves.